torch/_dynamo/guards.py

```diff
@@ -715,6 +715,7 @@ def TENSOR_MATCH(self, guard: Guard, value=None):
                     "device",
                     "requires_grad",
                     "ndimension()",
+                    "layout",
```

Comment:
@voznesenskym I hope this is ok? do you want a separate PR for this?

Reply:
@tugsbayasgalan is export even using these guards for anything lol

Reply:
This is fine.

Reply:
I can separate out for modularity (in case it breaks something downstream)

Reply:
removed for now